app/controllers/users/users.profile.server.controller.js

```diff
@@ -54,3 +54,66 @@ exports.update = function(req, res) {
 exports.me = function(req, res) {
 	res.json(req.user || null);
 };
+
+/**
+ * Record User's vote and update Article vote counts
+ */
+exports.vote = function(req, res) {
+	var article = req.article;
+	var user = req.user;
+	var newVoteDirection = req.params.direction;
+	var oldVoteDirection;
+	var upVotesAddend = 0;
+	var downVotesAddend = 0;
+	var voteIndex;
+
+	// look for existing vote
+	for (voteIndex = 0; voteIndex<user.votes.length; voteIndex++) {
+		if (article._id.equals(user.votes[voteIndex]._id)) {
```

Comment:
I originally had == or === but was not getting the results which I expected as the type of what's being compared is ObjectId.  I believe that I got the equals solution from here:

http://stackoverflow.com/questions/11060213/mongoose-objectid-comparisons-fail-inconsistently

A straight == (or ===) comparison will compare the two objects by reference, not value. So that will only evaluate to true if they both reference the very same instance.

Instead, you should be using the equals method of ObjectID to compare their values